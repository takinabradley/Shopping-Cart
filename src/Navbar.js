import {Link} from "react-router-dom";
export default function Navbar() {
  return (
    <div data-testid='navbar'>
      <Link to="/">Home</Link>
      <Link to="/shop">Shop</Link>
    </div>
  );
}
