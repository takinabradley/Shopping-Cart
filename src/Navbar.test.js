import React from "react";
import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'
import userEvent from "@testing-library/user-event";
import Navbar from './Navbar';
import { MemoryRouter } from 'react-router-dom'

it('renders', () => {
  const view = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )

  const navbar = screen.getByTestId('navbar')
  expect(navbar).toBeInTheDocument()
  expect(view).toMatchSnapshot()
})

it('contains links to Home and Shop pages', () => {
  const view = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )
  const homeLink = screen.getByText('Home')
  const shopLink = screen.getByText('Shop')
  expect(homeLink).toHaveAttribute('href', '/')
  expect(shopLink).toHaveAttribute('href', '/shop')
  expect(view).toMatchSnapshot()
})