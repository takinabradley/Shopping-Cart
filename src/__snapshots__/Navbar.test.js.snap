// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    data-testid="navbar"
  >
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/shop"
    >
      Shop
    </a>
  </div>
</div>
`;
